File: examples/javascript/08-runtime-export/runtime-export.mjs
// Gavio Runtime Export - metadata-safe JSONL event export.

import { Gateway, jsonlRuntimeExporter } from 'gavio'
import { piiGuard } from 'gavio/interceptors/pii'

const lines = []
const gateway = new Gateway({
  devMode: true,
  exporters: [jsonlRuntimeExporter({ write: (line) => lines.push(line) })],
}).use(piiGuard())

await gateway.complete({
  messages: [{ role: 'user', content: 'Email jan@example.com about ACME billing' }],
  metadata: { tenant: 'acme', feature: 'support-chat', environment: 'dev' },
})

const events = lines.map((line) => JSON.parse(line))
const contentKeys = ['messages', 'content', 'diff']
const leaked = events.filter((event) =>
  contentKeys.some((key) => JSON.stringify(event.data).includes(`"${key}"`)),
)

console.log(`exported_events=${events.length}`)
console.log(`event_types=${JSON.stringify(events.map((event) => event.type))}`)
console.log(`content_keys_exported=${leaked.length > 0}`)

File: examples/python/08-runtime-export/runtime_export.py
"""Gavio Runtime Export - metadata-safe JSONL event export."""

from __future__ import annotations

import asyncio
import io
import json

from gavio import Gateway, JsonlRuntimeExporter
from gavio.interceptors.pii import PiiGuard


async def main() -> None:
    stream = io.StringIO()
    gateway = (
        Gateway.builder()
        .dev_mode(True)
        .use(PiiGuard())
        .exporter(JsonlRuntimeExporter(stream=stream))
        .build()
    )

    await gateway.complete(
        messages=[{"role": "user", "content": "Email jan@example.com about ACME billing"}],
        metadata={"tenant": "acme", "feature": "support-chat", "environment": "dev"},
    )

    events = [json.loads(line) for line in stream.getvalue().splitlines()]
    content_keys = {"messages", "content", "diff"}
    leaked = [
        event["type"]
        for event in events
        if any(key in json.dumps(event["data"]) for key in content_keys)
    ]

    print(f"exported_events={len(events)}")
    print(f"event_types={[event['type'] for event in events]}")
    print(f"content_keys_exported={bool(leaked)}")


if __name__ == "__main__":
    asyncio.run(main())
